x-pack/plugins/secops/public/components/page/network/network_top_n_flow_table/columns.tsx

```diff
@@ -6,35 +6,38 @@
 
 import { EuiIcon, EuiToolTip } from '@elastic/eui';
 import numeral from '@elastic/numeral';
-import { get, isEmpty } from 'lodash/fp';
+import { get, isEmpty, isNil } from 'lodash/fp';
 import React from 'react';
 import styled from 'styled-components';
 
 import {
   NetworkDirectionEcs,
   NetworkTopNFlowDirection,
   NetworkTopNFlowEdges,
+  NetworkTopNFlowItem,
   NetworkTopNFlowType,
 } from '../../../../graphql/types';
 import { escapeQueryValue } from '../../../../lib/keury';
 import { networkModel } from '../../../../store';
 import { DragEffects, DraggableWrapper } from '../../../drag_and_drop/draggable_wrapper';
 import { escapeDataProviderId } from '../../../drag_and_drop/helpers';
-import { defaultToEmptyTag, getEmptyTagValue, getOrEmptyTag } from '../../../empty_value';
+import { defaultToEmptyTag, getEmptyTagValue } from '../../../empty_value';
 import { IPDetailsLink } from '../../../links';
 import { Columns } from '../../../load_more_table';
 import { Provider } from '../../../timeline/data_providers/provider';
 
 import { AddToKql } from './add_to_kql';
 import * as i18n from './translations';
 
+type valueof<T> = T[keyof T];
+
 export const getNetworkTopNFlowColumns = (
   startDate: number,
   topNFlowDirection: NetworkTopNFlowDirection,
   topNFlowType: NetworkTopNFlowType,
   type: networkModel.NetworkType,
   tableId: string
-): Array<Columns<NetworkTopNFlowEdges>> => [
+): Array<Columns<NetworkTopNFlowEdges | valueof<NetworkTopNFlowItem>>> => [
```

Comment:
The weirdness with typing like this is if you keep it as you have it, it looks like you end up with `({ node })` and everything below as a typed `any` now.

For example `{node}` as seen in my IDE is now an `any`:

<img width="613" alt="Screen Shot 2019-03-19 at 8 30 09 AM" src="https://user-images.githubusercontent.com/1151048/54614010-4f84f300-4a21-11e9-88f9-4ad4c400447e.png">

However, if I remove the `valueof` it returns it back to a non any

<img width="446" alt="Screen Shot 2019-03-19 at 8 30 00 AM" src="https://user-images.githubusercontent.com/1151048/54614066-64fa1d00-4a21-11e9-92e7-ea0a1f883160.png">

I didn't dive into the `why` you need the valueof type but assume it's needed.

Actionable suggestion is if you have to keep it to type the ({ node }) as in:

```ts
render: ({ node }: { node: NetworkTopNFlowItem }) => {
```

here and other spots like you have the other render calls.

Reply:
yes but if you remove the valueof, then the render is getting upset that's why I used `valueof` to get the type inside of `NetworkTopNFlowItem` like for direction, bytes, packets.
if you think that you have a better way please share because I did not really know what to do here